go/pvl/interp.go

```diff
@@ -138,29 +141,50 @@ func checkProofInner(g proofContextExt, pvl *jsonw.Wrapper, service keybase1.Pro
 			"Bad signature: %v", err)
 	}
 
-	scripts, perr := chunkGetScripts(pvl, service)
-	if perr != nil {
-		return perr
+	vars := scriptVariables{
+		UsernameService: info.RemoteUsername, // Empty "" for web and dns proofs
+		UsernameKeybase: info.Username,
+		Sig:             sigBody,
+		SigIDMedium:     sigID.ToMediumID(),
+		SigIDShort:      sigID.ToShortID(),
+		Hostname:        info.Hostname, // Empty "" except for web/dns proofs
+		Protocol:        info.Protocol, // Empty "" except for web proofs
 	}
 
-	newstate := func(i int) scriptState {
-		vars := scriptVariables{
-			UsernameService: info.RemoteUsername, // Blank for DNS-proofs
-			UsernameKeybase: info.Username,
-			Sig:             sigBody,
-			SigIDMedium:     sigID.ToMediumID(),
-			SigIDShort:      sigID.ToShortID(),
-			Hostname:        info.Hostname, // Blank for non-{DNS/Web} proofs
-		}
+	// Enforce prooftype-dependent variables.
+	webish := (service == keybase1.ProofType_DNS || service == keybase1.ProofType_GENERIC_WEB_SITE)
+	if webish {
+		vars.UsernameService = ""
+	} else {
+		vars.Hostname = ""
+	}
+	if service != keybase1.ProofType_GENERIC_WEB_SITE {
+		vars.Protocol = ""
+	}
```

Comment:
Should we be asserting that these variables weren't set at all, rather than clearing them out?